ambry-tools/src/main/java/com.github.ambry/store/StoreCopier.java

```diff
@@ -0,0 +1,246 @@
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.StaticClusterAgentsFactory;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.Config;
+import com.github.ambry.config.Default;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.BlobStoreRecovery;
+import com.github.ambry.messageformat.MessageFormatWriteSet;
+import com.github.ambry.tools.util.ToolUtils;
+import com.github.ambry.utils.ByteBufferChannel;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.ByteArrayInputStream;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Copies messages from one {@link Store} to another and provides the ability to transform the message along the way.
+ * Does not check whether the destination store fits all the messages from the source store - this check has to be
+ * made before copying
+ * <p/>
+ * This tool requires the source store to *not* return blobs that have already been deleted when
+ * {@link Store#findEntriesSince(FindToken, long)} is called. It is also expected to be run when both locations (src and
+ * tgt) are offline.
+ */
+public class StoreCopier implements Closeable {
+
+  /**
+   * Representation of a message in the store. Contains the {@link MessageInfo} and the {@link InputStream} of data.
+   */
+  public static class Message {
+    private final MessageInfo messageInfo;
+    private final InputStream stream;
+
+    /**
+     * @param messageInfo the {@link MessageInfo} for this message.
+     * @param stream the {@link InputStream} that represents the data of this message.
+     */
+    public Message(MessageInfo messageInfo, InputStream stream) {
+      this.messageInfo = messageInfo;
+      this.stream = stream;
+    }
+
+    /**
+     * @return the {@link MessageInfo} for this message.
+     */
+    public MessageInfo getMessageInfo() {
+      return messageInfo;
+    }
+
+    /**
+     * @return the {@link InputStream} that represents the data of this message.
+     */
+    public InputStream getStream() {
+      return stream;
+    }
+  }
+
+  /**
+   * An interface for a transformation function. Transformations can modify any data in the message (including keys).
+   */
+  public interface Transformer {
+    Message transform(Message message);
+  }
+
+  /**
+   * Config class for the {@link StoreCopier}.
+   */
+  static class CopierConfig {
+
+    /**
+     * The path of the directory where the source store files are.
```

Comment:
I didn't think that was necessary since these configs are not a part of the api - they are only for this tool and are used _only_ in the main function - a way to provide command line args. They aren't even used in the actual class/implementation.